qiskit_experiments/randomized_benchmarking/rb_analysis.py

```diff
@@ -0,0 +1,155 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Standard RB analysis class.
+"""
+
+from typing import Optional, List
+
+import numpy as np
+from qiskit_experiments.base_analysis import BaseAnalysis
+from qiskit_experiments.analysis.curve_fitting import curve_fit
+from qiskit_experiments.analysis.data_processing import (
+    level2_probability,
+    mean_xy_data,
+    filter_data,
+)
+from qiskit_experiments.analysis.plotting import plot_curve_fit
+
+try:
+    from matplotlib import pyplot as plt
+
+    HAS_MATPLOTLIB = True
+except ImportError:
+    HAS_MATPLOTLIB = False
+
+
+class RBAnalysis(BaseAnalysis):
+    """RB Analysis class."""
+
+    # pylint: disable = arguments-differ, invalid-name, attribute-defined-outside-init
+    def _run_analysis(
+        self,
+        experiment_data,
+        p0: Optional[List[float]] = None,
+        plot: bool = True,
+        ax: Optional["AxesSubplot"] = None,
+    ):
+        """Run analysis on circuit data.
+        Args:
+            experiment_data (ExperimentData): the experiment data to analyze.
+            p0: Optional, initial parameter values for curve_fit.
+            plot: If True generate a plot of fitted data.
+            ax: Optional, matplotlib axis to add plot to.
+        Returns:
+            tuple: A pair ``(analysis_result, figures)`` where
+                   ``analysis_results`` may be a single or list of
+                   AnalysisResult objects, and ``figures`` may be
+                   None, a single figure, or a list of figures.
+        """
+        self._num_qubits = len(experiment_data.data[0]["metadata"]["qubits"])
+        xdata, ydata, ydata_sigma = self._extract_data(experiment_data)
+
+        def fit_fun(x, a, alpha, b):
+            return a * alpha ** x + b
+
+        p0 = self._p0(xdata, ydata)
+        analysis_result = curve_fit(
+            fit_fun,
+            xdata,
+            ydata,
+            p0,
+            ydata_sigma,
+            bounds=([0, 0, 0], [1, 1, 1]),
+            absolute_sigma=False,
```

Comment:
Why is it important for you to override `absolute_sigma` to `False` ?

Reply:
This was used to avoid a bug in `curve_fit` which is now fixed, so I removed it.